### DISCUSSION TITLE
3d Classification using MIL

### DISCUSSION THREAD

**Discussion by Meddebma** [Project-MONAI/MONAI#3405] [Ideas]
Hi, 

I was trying to apply the newly proposed MIL Model to my classification task using head CT-scans, replacing Densenet121 with MILModel. Unfortunately it doesnt work with the loaded images. Should I change the parameters of the model or how should I proceed? Thank you very much

https://github.com/Meddebma/pyradiomics/blob/master/Classification_HBI.ipynb

**Comment by Nic-Ma** [comment]
Hi @Meddebma ,

Thanks for your feedback, I think the issue is related to mismatch input data shape or dims.
@myron Could you please help double confirm it?
I will try to enhance the doc-string to highlight the expected data shape of MIL module later.

Thanks in advance.

**Reply by myron** [reply]
Hi, you are initializing MILModel with defaults (which uses pretrained ResNet50, which expects a 3 channel input. And it seems your input has only 2 channels. I suggest you manually initialize a backbone CNN (resnet, densnet, etc) with the required number of input channels , and provide it as a backbone parameter to MILModel.
